apps/api/lib/validations/shared/querySlug.ts

```diff
@@ -0,0 +1,13 @@
+import { z } from "zod";
+
+import { baseApiParams } from "./baseApiParams";
+
+export const schemaQuerySlug = baseApiParams
+  .extend({
+    slug: z.string(),
+    ownerId: z
+      .string()
+      .optional()
+      .transform((value) => (value ? parseInt(value, 10) : null)),
+  })
+  .strict();
```

Comment:
I refactored this file in commit [refactor: schema query slug](https://github.com/calcom/cal.com/pull/13374/commits/9c6b51858d5252b2312438417a25671865f09ca2):
1. There is no need to include "String" in querySlugString.ts because slug is a string already, so i renamed file from "querySlugString.ts" to "querySlug.ts"
2. "export const withValidQuerySlugString" was not used anywhere in codebase so i deleted it
3. Admins should be able to access event types of other users, so optional "ownerId" query parameter has been added in commit [feat: allow admins to fetch event types on behalf of others](https://github.com/calcom/cal.com/pull/13374/commits/6765cc7b049607234b4440bbe436a1aa0931ac0d).